ytypes/container.go

```diff
@@ -100,46 +100,55 @@ func validateContainer(schema *yang.Entry, value ygot.GoStruct) (errors []error)
 //   schema is the schema of the schema node corresponding to the struct being
 //     unmamshaled into.
 //   parent is the parent struct, which must be a struct ptr.
-//   value is a JSON data tree which must be a map[string]interface{}.
-func unmarshalContainer(schema *yang.Entry, parent interface{}, value interface{}) error {
-	if util.IsValueNil(value) {
+//   jsonTree is a JSON data tree which must be a map[string]interface{}.
+func unmarshalContainer(schema *yang.Entry, parent interface{}, jsonTree interface{}) error {
+	if util.IsValueNil(jsonTree) {
 		return nil
 	}
 
-	util.DbgPrint("unmarshalContainer value %v, type %T, into parent type %T, schema name %s", util.ValueStr(value), value, parent, schema.Name)
+	util.DbgPrint("unmarshalContainer jsonTree %v, type %T, into parent type %T, schema name %s", util.ValueStr(jsonTree), jsonTree, parent, schema.Name)
 
 	// Check that the schema itself is valid.
 	if err := validateContainerSchema(schema); err != nil {
 		return err
 	}
 
 	// Since this is a container, the JSON data tree is a map.
-	jsonTree, ok := value.(map[string]interface{})
+	jt, ok := jsonTree.(map[string]interface{})
 	if !ok {
-		return fmt.Errorf("unmarshalContainer for schema %s: value %v: got type %T inside container, expect map[string]interface{}",
-			schema.Name, util.ValueStr(value), value)
+		return fmt.Errorf("unmarshalContainer for schema %s: jsonTree %v: got type %T inside container, expect map[string]interface{}",
+			schema.Name, util.ValueStr(jsonTree), jsonTree)
 	}
 
 	pvp := reflect.ValueOf(parent)
 	if !util.IsValueStructPtr(pvp) {
 		return fmt.Errorf("unmarshalContainer got parent type %T, expect struct ptr", parent)
 	}
-	pv := pvp.Elem()
 
+	return unmarshalStruct(schema, parent, jt)
+}
+
+// unmarshalStruct unmarshals a JSON tree into a struct.
+//   schema is the schema of the schema node corresponding to the struct being
```

Comment:
"schema is the YANG schema of the node corresponding to the struct being unmarshalled into"?

Reply:
Done.